### PR TITLE
Feature: select calendars to check for availability

### PR DESCRIPTION
**Pull request by Malte-D**
Implemented calendar selection for availability checking.
Also upgraded outlook integration to be able to check all calendars instead of only the default one.

### CODE REVIEW

**Comment by Malte-D** [comment]
I skipped the creation of a migration for now, because it would create duplicates with https://github.com/calendso/calendso/pull/261.
After that is merged, the migration can be added here

**Comment by baileypumfleet** [comment]
I've merged Nico's migrations PR here, so that should resolve the conflict

File: prisma/schema.prisma [hunk 1]
```diff
@@ -49,6 +49,7 @@ model User {
   credentials   Credential[]
   teams         Membership[]
   bookings      Booking[]
+  selectedCalendars      SelectedCalendar[]
```
Comment (emrysal) [inline prisma/schema.prisma:52]: I'm thinking, maybe it makes sense to just call this "Calendar"?

Comment (Malte-D) [reply prisma/schema.prisma:52]: With that name I wanted to express that the list does not contain a list of all available calendars (which would require some sort of syncing with all integrations), but only the selected ones.

File: lib/calendarClient.ts [hunk 2]
```diff
@@ -112,37 +121,59 @@ const MicrosoftOffice365Calendar = (credential): CalendarApiAdapter => {
...
-                            end: evt.end.dateTime + 'Z'
-                        }))
+                (accessToken) => {
+                    const selectedCalendarIds = selectedCalendars.filter(e => e.integration === integrationType).map(e => e.externalId);
+                    if (selectedCalendarIds.length == 0 && selectedCalendars.length > 0){
+                        // Only calendars of other integrations selected
+                        return Promise.resolve([]);
+                    }
+
+                    console.log("selectedCalendarIds.length: " + selectedCalendarIds.length)
```
Comment (Nico-J) [inline lib/calendarClient.ts:156]: Is this debugging log? If yes, we could leave it out.

File: lib/calendarClient.ts [hunk 3]
```diff
@@ -112,37 +121,59 @@ const MicrosoftOffice365Calendar = (credential): CalendarApiAdapter => {
...
+                    return (selectedCalendarIds.length == 0
+                      ? listCalendars().then(cals => cals.map(e => e.externalId))
+                      : Promise.resolve(selectedCalendarIds).then(x => x)).then((ids: string[]) => {
+                        const urls = ids.map(calendarId => 'https://graph.microsoft.com/v1.0/me/calendars/' + calendarId + '/events' + filter)
+                        console.log("urls", urls)
```
Comment (Nico-J) [inline lib/calendarClient.ts:161]: Same thing here

File: pages/integrations/index.tsx [hunk 4]
```diff
@@ -2,29 +2,82 @@ import Head from 'next/head';
...
+    function toggleShowCalendarModal() {
+        setShowSelectCalendarModal(!showSelectCalendarModal);
+    }
+
+    function loadCalendars() {
+        fetch('api/availability/calendar')
+          .then((response) => response.json())
+          .then(data => {
+              setSelectableCalendars(data)
+          });
```
Comment (Nico-J) [inline pages/integrations/index.tsx:30]: We should consider introducing a global error handler to catch rejected promises and display them to the user / catch and ignore them if that's fine. Or at least use `.catch((err) => do whatever you want with err);` somewhere in your Promise chain after `fetch(...)`. For now (not a stable version yet), I'd say it's fine as you did it, but I think for the future that would be neat. Or what do you think @baileypumfleet @emrysal @PeerRich?

**Review by Nico-J — APPROVED** [review]
Besides the .catch() thing on fetch Promises (which I cannot decide alone), the code is fine and you got my approvement. :)

**Comment by Malte-D** [comment]
@emrysal Did you find anything else?

**Pull request merged**